src-tauri/src/operations/node_operations.rs

```diff
@@ -79,85 +91,298 @@ pub fn get_nodes(state: State<'_, AppState>) -> Result<Vec<NodeInfo>> {
-  original_node_name: String,
-  node_name: String,
-  server_port: u32,
-  swarm_port: u32,
-  run_on_startup: bool,
+    state: State<'_, AppState>,
+    original_node_name: String,
+    node_name: String,
+    server_port: u32,
+    swarm_port: u32,
+    run_on_startup: bool,
 ) -> Result<OperationResult> {
-  let config_path = get_nodes_dir(&state.app_handle)
-    .join(&original_node_name)
-    .join("config.toml");
-  let config_content = fs::read_to_string(&config_path)?;
-
-  let mut config: Value = toml::from_str(&config_content)?;
-
-  // Modify the "swarm" section by changing the port numbers to 1234
-  if let Some(swarm) = config.get_mut("swarm") {
-      if let Some(listen) = swarm.get_mut("listen").and_then(|v| v.as_array_mut()) {
-          for entry in listen.iter_mut() {
-              if let Some(s) = entry.as_str() {
-                  // Replace the port number with 1234 in all entries
-                  let new_value = change_port_in_path(s, &swarm_port.to_string());
-                  *entry = Value::String(new_value); // Update the entry with a new string
-              }
-          }
-      }
-  }
-
-  // Modify the "server" section by changing the port numbers to 9876
-  if let Some(server) = config.get_mut("server") {
-      if let Some(listen) = server.get_mut("listen").and_then(|v| v.as_array_mut()) {
-          for entry in listen.iter_mut() {
-              if let Some(s) = entry.as_str() {
-                  // Replace the port number with 9876 in all entries
-                  let new_value = change_port_in_path(s, &server_port.to_string());
-                  *entry = Value::String(new_value); // Update the entry with a new string
-              }
-          }
-      }
-  }
-
-  // Serialize the updated config back to TOML
-  let updated_content = toml::to_string(&config)?;
-
-  // Write the updated content back to the file
-  fs::write(&config_path, updated_content)?;
-
-  update_run_node_on_startup(&state, &node_name, run_on_startup)?;
-
-  // Rename the node directory if the name has changed
-  if original_node_name != node_name {
-      let nodes_dir = get_nodes_dir(&state.app_handle);
-      let node_dir = nodes_dir.join(&original_node_name);
-      let new_node_dir = nodes_dir.join(&node_name);
-      fs::rename(&node_dir, &new_node_dir)?;
-  }
-
-  Ok(OperationResult {
-      success: true,
-      message: format!(
-          "Configuration for node '{}' has been updated successfully.",
-          node_name
-      ),
-  })
+    let config_path = get_nodes_dir(&state.app_handle)
+        .join(&original_node_name)
+        .join("config.toml");
+    let config_content = fs::read_to_string(&config_path)?;
+
+    let mut config: Value = toml::from_str(&config_content)?;
+
+    // Modify the "swarm" section by changing the port numbers to 1234
+    if let Some(swarm) = config.get_mut("swarm") {
+        if let Some(listen) = swarm.get_mut("listen").and_then(|v| v.as_array_mut()) {
```

Comment:
_... you can do `config["swarm"]["listen"]`

but ideally you should be using `toml_edit` which affords you the same thing without reformatting the file, up to you though (since it's an app-managed config it doesn't matter much)

we already do this for `meroctl config`: https://github.com/calimero-network/core/blob/9f7b6fe3d3552e04443c89f21784d0dd1ce5d569/crates/meroctl/src/cli/config.rs#L98-L100

would've been nice if you could use [`ConfigFile`](https://github.com/calimero-network/core/blob/9f7b6fe3d3552e04443c89f21784d0dd1ce5d569/crates/meroctl/src/config_file.rs#L16)::[`load()`](https://github.com/calimero-network/core/blob/9f7b6fe3d3552e04443c89f21784d0dd1ce5d569/crates/meroctl/src/config_file.rs#L77), but right now that's not contained in a primitive, so it won't be clean